internal/k8s/reconciler/utils.go

```diff
@@ -121,35 +121,6 @@ func routeAllowedForListenerNamespaces(ctx context.Context, gatewayNS string, al
 	return false, nil
 }
 
-// gatewayAllowedForSecretRef determines whether the gateway is allowed
-// for the secret either by being in the same namespace or by having
-// an applicable ReferenceGrant in the same namespace as the secret.
-func gatewayAllowedForSecretRef(ctx context.Context, gateway *gwv1beta1.Gateway, secretRef gwv1beta1.SecretObjectReference, c gatewayclient.Client) (bool, error) {
```

Comment:
This is only used during the validation process that determines what the `GatewayState` should be, so it's moved into the `validator` package